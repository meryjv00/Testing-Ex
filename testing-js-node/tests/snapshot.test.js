/* eslint-disable no-undef */
// Prueba de Snapshot
var geoObject = require('../src/snapshot/datos.db.json');
var enlace = require('../src/snapshot/enlance.json');
const snapshot = require('../src/snapshot/snapshot');

// Creamos el snapshopt
describe('Snapshot', () => {
  // Prueba de Snapshot
  test('Prueba de snapshot', () => {
    // De esta manera siempre comparamos lo que nos llega con lo que hay en la carpeta __snapshots__ (se crea la primera vez)
    expect(snapshot.snapShotObject(geoObject)).toMatchSnapshot();
  });
  // Simulamos que el enlace sea el mismo que el almacenando
  test('Simular UI', () => {
    expect(snapshot.href(enlace)).toMatchSnapshot();
  });

  // Test con cambios permitidos
  test('Log de actividad', () => {
    const log = {
      id: 1,
      date: new Date(),
      active: Math.floor(Math.random() * 100) + 1 < 50,
    };
    // expect(log).toMatchSnapshot();
    expect(log).toMatchSnapshot({
      date: expect.any(Date),
      active: expect.any(Boolean),
    });
  });
});

// Repite el test pero cambia los ficheros originales, como no coiciden con la instantánea, pues dará error
