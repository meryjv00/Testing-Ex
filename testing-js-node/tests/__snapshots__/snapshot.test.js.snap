// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Log de actividad 1`] = `
Object {
  "active": Any<Boolean>,
  "date": Any<Date>,
  "id": 1,
}
`;

exports[`Snapshot Prueba de snapshot 1`] = `
Object {
  "active": true,
  "id": 1,
  "latitude": "3.5645543",
  "longitude": "40.232332",
}
`;

exports[`Snapshot Simular UI 1`] = `
Object {
  "link": "<a href='https://www.google.es' class='button'>Enlace</a>",
}
`;
